packages/core/src/connectors/metaMask.ts

```diff
@@ -74,12 +92,17 @@ export class MetaMaskConnector extends InjectedConnector {
         const isConnected = !!account
         if (isConnected)
           // Attempt to show another prompt for selecting wallet if already connected
-          await provider
-            .request({
+          try {
+            await provider.request({
               method: 'wallet_requestPermissions',
               params: [{ eth_accounts: {} }],
             })
-            .catch(() => null)
+          } catch (error) {
+            // Not all MetaMask injected providers support `wallet_requestPermissions` (e.g. MetaMask iOS).
+            // Only bubble up error if user rejects request
+            if (this.isUserRejectedRequestError(error))
+              throw new UserRejectedRequestError(error)
+          }
```

Comment:
#1384 broken this logic by swallowing all errors so if you rejected a request, you were still connected. Only affected folks with `UNSTABLE_shimOnConnectSelectAccount` turned on.